src/data/real/nnreal.lean

```diff
@@ -725,6 +759,98 @@ end inv
 @[simp] lemma abs_eq (x : ℝ≥0) : |(x : ℝ)| = x :=
 abs_of_nonneg x.property
 
+section csupr
+open set
+
+variables {ι : Sort*} {f : ι → ℝ≥0}
+
+lemma le_to_nnreal_of_coe_le {x : ℝ≥0} {y : ℝ} (h : ↑x ≤ y) : x ≤ y.to_nnreal :=
+(le_to_nnreal_iff_coe_le $ x.2.trans h).2 h
+
+lemma supr_of_not_bdd_above (hf : ¬ bdd_above (range f)) : (⨆ i, f i) = 0 :=
+begin
+  ext,
+  rw coe_supr,
+  refine supr_of_not_bdd_above _,
+  rintro ⟨r, hr⟩,
+  exact hf ⟨r.to_nnreal, forall_range_iff.2 $ λ i, le_to_nnreal_of_coe_le $ hr $ mem_range_self _⟩,
```

Comment:
Could you please add (or use if it exists) the following lemma? One implication is `monotone.map_bdd_above`, another is the last line of your proof.
```
lemma nnreal.bdd_above_image_coe {s : set ℝ≥0} :
  bdd_above (coe '' s : set ℝ) ↔ bdd_above s
```

Then you can prove `Sup_of_not_bdd_above` and `supr_of_not_bdd_above`.